p2p/pex/addrbook_test.go

```diff
@@ -36,6 +37,212 @@ func deleteTempFile(fname string) {
 	}
 }
 
+func TestAddrBookGetSelectionWithOneMarkedGood(t *testing.T) {
+	// create a book with 10 addresses, 1 good/ old and 9 new
+	book := createAddrBookWithMOldAndNNewAddrs(t, 1, 9)
+
+	// ensure we can get selection.
+	// this used to result in an infinite loop (!)
+	addrs := book.GetSelectionWithBias(biasToSelectNewPeers)
+	assert.NotNil(t, addrs, "expected an address")
```

Comment:
We should be able to assert exactly how many we get now, and in the next test.

Reply:
added